elide-async/src/main/java/com/yahoo/elide/async/service/thread/AsyncAPICallable.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service.thread;
+
+import com.yahoo.elide.async.models.AsyncAPI;
+import com.yahoo.elide.async.models.AsyncAPIResult;
+import com.yahoo.elide.async.operation.AsyncAPIOperation;
+import com.yahoo.elide.core.RequestScope;
+import org.apache.http.NoHttpResponseException;
+import lombok.extern.slf4j.Slf4j;
+
+import java.net.URISyntaxException;
+import java.util.concurrent.Callable;
+
+/**
+ * AsyncAPI implementation of Callable for executing the query provided in AsyncQuery.
+ * It will also update the query status and result object at different stages of execution.
+ */
+@Slf4j
+public class AsyncAPICallable implements Callable<AsyncAPIResult> {
```

Comment:
Maybe we should merge this with the Operation (make the operation extend callable) since it doesn't do anything but call the operation.